apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/FirestoreDb.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright 2017, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.using System;
+
+using Google.Api.Gax;
+using Google.Cloud.Firestore.V1Beta1;
+using System;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Firestore.Data
+{
+    // TODO: Abstract with all the trimmings?
```

Comment:
Yes, I think so. Sorry.

Reply:
Actually, we might want to revisit that idea, as there may be things we want to keep internal. Let's see how we go.